### PR TITLE
LPS-165376 - Creation of the function to update and delete user roles

### PR DESCRIPTION
**Pull request by caiosouza15**

### CODE REVIEW

**Comment by caiosouza15** [comment]
ci:test:sf

**Comment by caiosouza15** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx [hunk 1]
```diff
@@ -56,12 +60,30 @@ const UserForm = () => {
...
+		if (!rolesWatch?.length) {
+			hasError = true;
+
+			Liferay.Util.openToast({
+				message: i18n.translate('please-select-one-or-more-roles'),
+				type: 'danger',
+			});
+		}
+		if (hasError) {
+			return false;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx:84]: No need to return false

```suggestion
			return;
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx [hunk 2]
```diff
@@ -56,12 +60,30 @@ const UserForm = () => {
...
+		}
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx:85]: Simplify 

```suggestion
		if (!rolesWatch?.length) {
			return Liferay.Util.openToast({
				message: i18n.translate('please-select-one-or-more-roles'),
				type: 'danger',
			});
		}
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx [hunk 3]
```diff
@@ -71,15 +93,48 @@ const UserForm = () => {
...
 	};
-
-	const userRoles =
-		userAccount?.roleBriefs.map(({id}: {id: number}) => id) || [];
 	const roles = data?.items || [];
 
+	const checkPermissionRoles = roles.map((role: any) => {
+		return role.actions['create-role-user-account-association']
+			? true
+			: false;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx:112]: Simplify

```suggestion
		return !!role.actions['create-role-user-account-association'];
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx [hunk 4]
```diff
@@ -191,16 +246,27 @@ const UserForm = () => {
...
+								{id, name}: {id: number; name: string},
+								index: number
+							) => (
+								<div className="mt-2" key={id}>
+									<ClayCheckbox
+										checked={rolesWatch.includes(id)}
+										disabled={
+											checkPermissionRoles[index]
+												? false
+												: true
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx:260]: Simplify

```suggestion
											!checkPermissionRoles[index]
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@caiosouza15 it looks good, thank you 🎉 

Can you just do my suggestions changes?

**Comment by caiosouza15** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**